adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -455,18 +457,24 @@ private AuthenticationResult postMessage(String requestMessage, Map<String, Stri
             HttpWebResponse response = mWebRequestHandler.sendPost(authority, headers,
                     requestMessage.getBytes(AuthenticationConstants.ENCODING_UTF8),
                     "application/x-www-form-urlencoded");
+            final Map<String, List<String>> responseHeaders = response.getResponseHeaders();
```

Comment:
response headers are already parsed in processTokenResponse. can you move this code there, it has the httpEvent object there?

Reply:
Done 👍